master/internal/core_experiment.go

```diff
@@ -293,7 +293,13 @@ func (m *Master) parseCreateExperiment(ctx context.Context, req *apiv1.CreateExp
 	}
 	workspaceID := resolveWorkspaceID(workspaceModel)
 	isSingleNode := resources.IsSingleNode() != nil && *resources.IsSingleNode()
-	poolName, _, err := m.ResolveResources(resources.ResourcePool(), resources.SlotsPerTrial(), workspaceID, isSingleNode)
+	isUnmanaged := req.Unmanaged != nil && *req.Unmanaged
+	slotsPerTrial := resources.SlotsPerTrial()
+	if isUnmanaged {
+		slotsPerTrial = 0
+	}
+
+	poolName, _, err := m.ResolveResources(resources.ResourcePool(), slotsPerTrial, workspaceID, isSingleNode)
```

Comment:
is there any reason to call resolve resources at all if the experiment is unmanaged? e.g., we have a very similar bug if someone passes a non-existent resource pool.

Reply:
i guess what makes this one so painful is that the default is 1 slot which doesn't work

Reply:
+1, haran hit this recently as well: we don't need to validate resource pool names for unmanaged.

Reply:
Hi @stoksc and @ioga , I've moved `ResolveResources` and `TaskContainerDefaults` behind unmanaged condition